quickwit/quickwit-codegen/quickwit-codegen-example/src/codegen/hello.rs

```diff
@@ -92,6 +95,69 @@ impl tower::Service<HelloRequest> for HelloClient {
         Box::pin(fut)
     }
 }
+/// A tower block is a set of towers. Each tower is stack of layers (middlewares) that are applied to a service.
+#[derive(Debug)]
+struct HelloTowerBlock {
+    hello_svc: quickwit_common::tower::BoxService<
+        HelloRequest,
+        HelloResponse,
+        crate::HelloError,
+    >,
+}
+impl Clone for HelloTowerBlock {
+    fn clone(&self) -> Self {
+        Self {
+            hello_svc: self.hello_svc.clone(),
+        }
+    }
+}
+#[async_trait::async_trait]
+impl Hello for HelloTowerBlock {
+    async fn hello(
+        &mut self,
+        request: HelloRequest,
+    ) -> crate::HelloResult<HelloResponse> {
+        self.hello_svc.ready().await?.call(request).await
+    }
+}
+#[derive(Debug, Default)]
+pub struct HelloTowerBlockBuilder {
+    hello_layer: Option<
+        quickwit_common::tower::BoxLayer<
+            Box<dyn Hello>,
+            HelloRequest,
+            HelloResponse,
+            crate::HelloError,
+        >,
+    >,
+}
+impl HelloTowerBlockBuilder {
+    pub fn hello_layer(
+        mut self,
+        layer: quickwit_common::tower::BoxLayer<
```

Comment:
This is where I could be happier about the API. I wanted layer build methods to accept `*_layer<S>(layer: Layer<S>) where ...` and perform the boxing inside the functions. Unfortunately, I could not do so while making the compiler happy, and I gave up, given the pressing need to ship this feature for the next release.

Anybody, feel free to take a stab.